ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/upload/HDFSUploadClient.java

```diff
@@ -85,17 +84,16 @@ public void init(LogFeederProps logFeederProps) {
     }
     logger.info("HDFS client - will use '{}' permission for uploaded files", hdfsOutputConfig.getHdfsFilePermissions());
     LogFeederHDFSUtil.overrideFileSystemConfigs(logFeederProps, configuration);
-    this.fs = LogFeederHDFSUtil.buildFileSystem(configuration);
   }
 
   @Override
   public void upload(String source, String target) throws Exception {
+    final FileSystem fs = LogFeederHDFSUtil.buildFileSystem(configuration);
```

Comment:
Should this be a try..finally block with close?

Reply:
could be but if the filesystem is not accessible it blocks the thread, so it would not be required, but it would be nice if we could call that on the actual used object during shutdown